crates/uv-resolver/src/resolver/mod.rs

```diff
@@ -267,18 +267,25 @@ impl<Provider: ResolverProvider, InstalledPackages: InstalledPackagesProvider>
         let solver = state.clone();
         let index_locations = provider.index_locations().clone();
         let (tx, rx) = oneshot::channel();
-        thread::Builder::new()
+        let handle = thread::Builder::new()
             .name("uv-resolver".into())
             .spawn(move || {
                 let result = solver.solve(index_locations, request_sink);
-                tx.send(result).unwrap();
+                tx.send(result)
             })
             .unwrap();
 
         let resolve_fut = async move { rx.await.map_err(|_| ResolveError::ChannelClosed) };
 
         // Wait for both to complete.
         let ((), resolution) = tokio::try_join!(requests_fut, resolve_fut)?;
+
+        // Retrieve a possible send error from the solver thread
+        handle
+            .join()
+            .unwrap()
+            .map_err(|_| ResolveError::ChannelClosed)?;
```

Comment:
But, will the error message here be different when we hit this case?

Reply:
I'm not sure — but the only error case that can be raised here is `send` failing. I'm hoping that the panic was just interrupting proper handling of the HTTP error in another task / thread.

Reply:
We could test this perhaps by manually throwing an HTTP error? I couldn't reproduce the flake locally though.

Reply:
I don't really understand why this would help. The solver thread was previously detached, so any panics shouldn't crash uv?